compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -716,17 +727,20 @@ impl<'a> Parser<'a> {
     }
 
     /// Is the token `dist` away from the current suitable as the start of a range patterns end?
-    fn is_pat_range_end_start(&self, dist: usize) -> bool {
+    fn is_pat_range_end_start(&mut self, dist: usize) -> bool {
         self.look_ahead(dist, |t| {
-            t.is_path_start() // e.g. `MY_CONST`;
-                || t.kind == token::Dot // e.g. `.5` for recovery;
-                || t.can_begin_literal_maybe_minus() // e.g. `42`.
+            t.is_inline_const_start()
+            || t.is_path_start() // e.g. `MY_CONST`;
+            || t.kind == token::Dot // e.g. `.5` for recovery;
+            || t.can_begin_literal_maybe_minus() // e.g. `42`.
                 || t.is_whole_expr()
         })
     }
 
     fn parse_pat_range_end(&mut self) -> PResult<'a, P<Expr>> {
-        if self.check_path() {
+        if self.check_inline_const() {
+            self.parse_const_expr(self.token.span)
```

Comment:
Semi-related: could you rename `parse_const_expr` into `parse_const_block`, this is a missing part of my old comment - https://github.com/rust-lang/rust/pull/77124#discussion_r494532128.